src/browser/components/SelectCertificateModal.jsx

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {Fragment} from 'react';
+import PropTypes from 'prop-types';
+import {Modal, Button, Table, Row, Col} from 'react-bootstrap';
+
+import ShowCertificateModal from './showCertificateModal.jsx';
+
+const CELL_SIZE = 23;
+const ELIPSIS_SIZE = 3;
+
+export default class SelectCertificateModal extends React.Component {
+  static propTypes = {
+    onSelect: PropTypes.func.isRequired,
+    onCancel: PropTypes.func,
+    certificateRequest: PropTypes.arrayOf(PropTypes.shape({
+      server: PropTypes.string,
+      certificateList: PropTypes.array,
+    })),
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedIndex: null,
+      showCertificate: null,
+    };
+  }
+
+  maxSize = (item, max) => {
+    if (!item || item.length <= max) {
+      return item;
+    }
+    const sub = item.substring(0, max - ELIPSIS_SIZE);
+    return `${sub}...`;
+  }
```

Comment:
FWIW, this can be accomplished automatically with CSS. No need to do anything now, but lets look at it once this is merged and you're doing a final CSS cleanup pass!